classes/phing/tasks/ext/ioncube/IoncubeEncoderTask.php

```diff
@@ -574,12 +574,7 @@ public function main()
     {
         $arguments = $this->constructArguments();
 
-        if (in_array($this->phpVersion, [5, 53, 54, 55, 56, 71])) {
-            $encoderName = $this->encoderName . $this->phpVersion;
-        } else {
-            $encoderName = $this->encoderName;
-        }
-        $encoder = new PhingFile($this->ioncubePath, $encoderName);
+        $encoder = new PhingFile($this->ioncubePath, $this->encoderName . $this->phpVersion);
```

Comment:
@MaximilianKresse thanks for your contribution. One small suggestion: would you mind to remove the default value of the `$this->phpVersion` with the actual running php version?
```php
// ...
    private $phpVersion;
// ...
        $encoder = new PhingFile(
            $this->ioncubePath,
            $this->encoderName . $this->phpVersion ?? str_replace('.', '', $this->getProject()->getProperty('php.version'))
        );
```
What do you think?

Reply:
This sounds great but in practice it doesn't work. The variable name "phpVersion" doesn't tell the whole truth - actually its the encoder version. The ioncube encoder has multiple encoder versions and additionally encoders for specific php versions:

![grafik](https://user-images.githubusercontent.com/545671/49697797-07ea0200-fbbc-11e8-85b6-b66912b905ad.png)
Note that there isn't a php 7 or 70 version so a direct mapping wouldn't work. Not every loader is compatible to every ioncube encoder. 

At the moment we always have to rename the files to match the expected shema of phing. With this change we could leave this ioncube encoder file names and change the "phpVersion" to something like "72_10.2_64".

And additionally I want to have so less logic as possible in this task because we don't know when ioncube likes to change there file structure ;-)